tests/iter.rs

```diff
@@ -0,0 +1,45 @@
+use std::{
+    future::Future,
+    pin::{pin, Pin},
+    task::{Context, Poll},
+};
+
+use futures_util::stream;
+use streamunordered::StreamUnordered;
+
+// An `async move { i }` causes unpin issues with `iter_mut` and `iter_mut_with_token`.
```

Comment:
You can just use [`std::future::ready`](https://doc.rust-lang.org/std/future/fn.ready.html) which also returns a future type that you can name and that is `Unpin` :+1: